modules/simple_forms_api/app/services/simple_forms_api/notification_email.rb

```diff
@@ -100,10 +100,12 @@ def send(at: nil)
     private
 
     def check_missing_keys(config)
-      missing_keys = %i[form_data form_number confirmation_number date_submitted].select { |key| config[key].nil? }
-      if config[:form_number] == 'vba_21_0966_intent_api' && config[:expiration_date].nil?
-        missing_keys << :expiration_date
-      end
+      all_keys = %i[form_data form_number date_submitted]
+      all_keys << :confirmation_number if needs_confirmation_number?(config)
+      all_keys << :expiration_date if config[:form_number] == 'vba_21_0966_intent_api'
+
+      missing_keys = all_keys.select { |key| config[key].nil? }
```

Comment:
Do you expect that there could be any empty strings here? You might want to check for nil and empty values. `{ |key| config[key].nil? || config[key].to_s.strip.empty? }`

Reply:
Great catch! Updated the PR. Thanks!